Add practice exercise `sgf-parsing` (#501)
exercism/elm: exercises/practice/sgf-parsing/.docs/instructions.md

  - The top level node has a single child which has a single property:
    B\[aa\].  (Black plays on the point encoded as "aa", which is the
    1-1 point).
    - The B\[aa\] node has a single child which has a single property:
      W\[ab\].

As you can imagine an SGF file contains a lot of nodes with a single
child, which is why there's a shorthand for it.

SGF can encode variations of play. Go players do a lot of backtracking
in their reviews (let's try this, doesn't work, let's try that) and SGF
supports variations of play sequences. For example:

```text
(;FF[4](;B[aa];W[ab])(;B[dd];W[ee]))
```

Here the root node has two variations. The first (which by convention
indicates what's actually played) is where black plays on 1-1. Black was
sent this file by his teacher who pointed out a more sensible play in
the second child of the root node: `B[dd]` (4-4 point, a very standard
opening to take the corner).

A key can have multiple values associated with it. For example:

```text
(;FF[4];AB[aa][ab][ba])
```

Here `AB` (add black) is used to add three black stones to the board.

There are a few more complexities to SGF (and parsing in general), which

> you can mostly ignore. You should assume that the input is encoded in

Author: I can't change the instructions, as they are centralized in the problem-specs repo.

> UTF-8, the tests won't contain a charset property, so don't worry about
> that. Furthermore you may assume that all newlines are unix style (`\n`,

Reviewer: ```suggestion
UTF-8 and the tests won't contain a charset property. Furthermore you may assume that all newlines are unix style (`\n`,
```

Author: same as above.
